packages/code-snippet/src/CodeSnippetWidget.tsx

```diff
@@ -247,10 +251,36 @@ class CodeSnippetDisplay extends MetadataDisplay<ICodeSnippetDisplayProps> {
     );
   }
 
+  static matchesSearch(
+    searchValue: string,
+    filterTags: Set<string>,
+    metadata: IMetadata
+  ): boolean {
+    // True if search string is in name, display_name, or language of snippet
+    // or if the search string is empty
+    const matchesSearch =
+      metadata.name.toLowerCase().includes(searchValue) ||
```

Comment:
I added the extra method `matchesSearch` so that if a child class of `MetadataWidget` (in this case, `CodeSnippetWidget`) wanted to match extra criteria in the search (in this case, the `language` field) they could override the default (which just matches the `name` or the `display_name`.) I'm assuming that the child class won't override the method that calls `matchesSearch` (`getDerivedStateFromProps`) and including the checks for things like empty strings and making the `searchValue` lowercase so that when a child class implements `matchesSearch` it won't have to use duplicate code. Does that make sense?

Reply:
I see.  I didn;t realize these were subclasses.  If that's the case, shouldn't this implementation call super.matchesSearch and, if no match, then check `language`?  However, because of the 'anded' behavior with tags, it seems like the two forms should be split (if you want to support subclassing like this).

I would suggest something like the following:
In the base class, implement `matchesSearch()` or whatever the primary name is.  This implementation calls two methods, one that matches via the metadata fields, let's say, `matchesMetadata()` and the other that matches tags, let's say, `matchesTags()`.  So you will have this in the base class (using python syntax):

```python
def matchesSearch(...):
    return matchesMetadata() and matchesTags()
```
Then, CodeSnippets would override `matchesMetadata()` with something like the following...

```python
def matchesMetadata():
    matches = super.matchesMetadata()
    if not matches:
        matches = <perform matching logic again `language` and any other code-snippet-only fields>
    return matches
```
Since the implementation for the tags matching is essentially metadata-type-independent, it will likely rarely be overridden.  LIkewise, the primary method, `matchesSearch()` would also rarely be overridden.

Reply:
as I discussed with @marthacryan , this is a great idea. Having separate `matchesSearch` and `matchesTags` will allow for better extensibility. Similarly about calling `super` rather than reimplement that functionality

Reply:
though it's called in the function that calls this, I feel that `toLower` should still be called  on `searchValue` here. A function should not make assumptions about what other functions that call it do with the params (unless absolutely necessary)